demos/common/cpp/models/src/detection_model_retinaface_pt.cpp

```diff
@@ -0,0 +1,250 @@
+    }
+    InferenceEngine::InputInfo::Ptr& input = inputInfo.begin()->second;
+    std::string imageInputName = inputInfo.begin()->first;
+    inputsNames.push_back(imageInputName);
+    input->setPrecision(InferenceEngine::Precision::U8);
+    if (useAutoResize) {
+        input->getPreProcess().setResizeAlgorithm(InferenceEngine::ResizeAlgorithm::RESIZE_BILINEAR);
+        input->getInputData()->setLayout(InferenceEngine::Layout::NHWC);
+    }
+    else {
+        input->getInputData()->setLayout(InferenceEngine::Layout::NCHW);
+    }
+
+    //--- Reading image input parameters
+    imageInputName = inputInfo.begin()->first;
+    const InferenceEngine::TensorDesc& inputDesc = inputInfo.begin()->second->getTensorDesc();
+    netInputHeight = getTensorHeight(inputDesc);
+    netInputWidth = getTensorWidth(inputDesc);
+
+    // --------------------------- Prepare output blobs -----------------------------------------------------
+    slog::info << "Checking that the outputs are as the demo expects" << slog::endl;
+
+    InferenceEngine::OutputsDataMap outputInfo(cnnNetwork.getOutputsInfo());
+    landmarksNum = 0;
+
+    std::vector<uint32_t> outputsSizes[OT_MAX];
+    for (auto& output : outputInfo) {
+        output.second->setPrecision(InferenceEngine::Precision::FP32);
+        output.second->setLayout(output.second->getDims().size()==4 ? InferenceEngine::Layout::NCHW : InferenceEngine::Layout::CHW);
+        outputsNames.push_back(output.first);
+
+        outputsNames.resize(2);
+        if (output.first.find("bbox") != std::string::npos) {
+            outputsNames[OT_BBOX] = output.first;
+        }
+        else if (output.first.find("cls") != std::string::npos) {
+            outputsNames[OT_SCORES] = output.first;
+        }
+        else if (output.first.find("landmark") != std::string::npos) {
+            outputsNames.resize(std::max(outputsNames.size(), (size_t)OT_LANDMARK + 1));
+            outputsNames[OT_LANDMARK] = output.first;
+            landmarksNum = output.second->getDims()[2]/2; // Each landmark consist of 2 variables (x and y)
+        }
+        else {
+            continue;
+        }
+    }
+
+    if (outputsNames[OT_BBOX] == "" || outputsNames[OT_SCORES] == "") {
+        throw std::runtime_error("Bbox or cls layers are not found");
+    }
+
+    if (outputsNames.size() != 2 && outputsNames.size() != 3) {
+        throw std::logic_error("Expected 2 or 3 output blobs");
+    }
+
+    priors = generatePriorData();
+}
+
+
+std::vector<uint32_t> ModelRetinaFacePT::doThresholding(const InferenceEngine::MemoryBlob::Ptr& rawData, const float confidenceThreshold) {
+    std::vector<uint32_t> indicies;
+    auto desc = rawData->getTensorDesc();
+    auto sz = desc.getDims();
+    InferenceEngine::LockedMemory<const void> outputMapped = rawData->rmap();
+    const float *memPtr = outputMapped.as<float*>();
+
+    for (uint32_t x = 0; x < sz[1]; ++x) {
+        auto idx = (x * sz[2] + 1);
```

Comment:
I'd prefer named constant for sz[1] and sz[2] values.
*Hereinafter*

Reply:
I will make code more complex. Generally, we have CHW layout for outputs, but values there are not H/W dimensions of image or something, but just general rows/columns count. I may put constants like SZ_ROWS and SZ_COLS but they will not give much explanation as they have different meaning in every functions (particularly, COLS may be equal to number of box coordinates, or to number of anchors multipliesd by point coordinates, etc...) . So I'd keep numeric indicies here - at least it will be consistent with model outputs layouts in model documentation.